bzt/modules/__init__.py

```diff
@@ -61,14 +61,33 @@ def reporting_setup(self, prefix=None, suffix=None):
             self.engine.aggregator.add_underling(self.reader)
 
 
-class SubprocessedExecutor(ReportableExecutor, FileLister, SelfDiagnosable, WidgetProvider):
+class IterationProvider(object):
+    def __init__(self):
+        self._start_handlers = []
+        self._end_handlers = []
+
+    def subscribe_to_iterations(self, start_handler, end_handler):
```

Comment:
This looks like unusual design for pub/sub pattern. Look at `MonitoringListener` or at `AggregatorListener` mixins. You should use same approach, instead of mixing listener and provider in single class.